va/va.go

```diff
@@ -707,7 +714,12 @@ func (va *ValidationAuthorityImpl) PerformValidation(ctx context.Context, req *v
 		return nil, errors.New("Challenge failed to deserialize")
 	}
 
-	records, err := va.validate(ctx, identifier.DNSIdentifier(req.Domain), req.Authz.RegID, challenge)
+	keyAuthorization := req.ExpectedKeyAuthorization
+	if len(keyAuthorization) == 0 {
+		keyAuthorization = req.Challenge.KeyAuthorization
+	}
```

Comment:
This should have another belt-and-suspenders check: if the keyAuthorization is _still_ empty, error.